pkg/config/agentmanagement.go

```diff
@@ -133,6 +135,13 @@ func (r remoteConfigHTTPProvider) FetchRemoteConfig() ([]byte, error) {
 		HTTPClientConfig: httpClientConfig,
 	}
 
+	if r.InitialConfig.RemoteConfiguration.LabelManagementEnabled && r.InitialConfig.RemoteConfiguration.AgentID != "" {
```

Comment:
shall we add some validations to the config to make sure that:

- no labels are defined when `label_management_enabled: true`
- `agent_id` is defined when  `label_management_enabled: true`

?